File: src/0880.py
class Solution:
    def decodeAtIndex(self, s: str, k: int) -> str:
        index = length = 0
        while length < k:
            if s[index].isdigit():
                length *= int(s[index])
            else:
                length += 1
            index += 1
        for index in range(index - 1, -1, -1):
            if s[index].isdigit():
                length //= int(s[index])
                k %= length
            else:
                if k == 0 or k == length:
                    return s[index]
                length -= 1

File: src/0880.cpp
#include <iostream>

using namespace std;

class Solution {
public:
    string decodeAtIndex(string s, int k) {
        long long length = 0;
        int index = 0;
        while(length < k) {
            if(isdigit(s[index])) {
                length *= (s[index]) - '0';
            }else {
                length++;
            }
            index++;
        }
        for(int j = index - 1; j >= 0; --j) {
            if(isdigit(s[j])) {
                length /= s[j] - '0';
                k %= length;
            }else {
                if(k == 0 || k == length) {
                    return string(1, s[j]);
                }
                length--;
            }
        }
        return "";
    }
};
